core/server/src/test/java/alluxio/worker/netty/DataServerWriteHandlerTest.java

```diff
@@ -0,0 +1,137 @@
+  protected EmbeddedChannel mChannelNoException;
+
+  /** The file used to hold the data written by the test. */
+  protected String mFile;
+
+  @Rule
+  public TemporaryFolder mTestFolder = new TemporaryFolder();
+
+  @Test
+  public void writeEmptyFile() throws Exception {
+    mChannel.writeInbound(buildWriteRequest(0, 0));
+
+    Object writeResponse = waitForResponse(mChannel);
+    checkWriteResponse(writeResponse, Protocol.Status.Code.OK);
+  }
+
+  @Test
+  public void writeNonEmptyFile() throws Exception {
+    long len = 0;
+    for (int i = 0; i < 128; i++) {
+      mChannel.writeInbound(buildWriteRequest(len, PACKET_SIZE));
+      len += PACKET_SIZE;
+    }
+    // EOF.
+    mChannel.writeInbound(buildWriteRequest(len, 0));
+
+    Object writeResponse = waitForResponse(mChannel);
+    checkWriteResponse(writeResponse, Protocol.Status.Code.OK);
+    checkFileContent(len);
+  }
+
+  @Test
+  public void writeInvalidOffset() throws Exception {
+    mChannelNoException.writeInbound(buildWriteRequest(0, PACKET_SIZE));
+    mChannelNoException.writeInbound(buildWriteRequest(PACKET_SIZE + 1, PACKET_SIZE));
+    Object writeResponse = waitForResponse(mChannelNoException);
+    Assert.assertTrue(writeResponse instanceof RPCProtoMessage);
+    checkWriteResponse(writeResponse, Protocol.Status.Code.INVALID_ARGUMENT);
+  }
+
+  /**
+   * Checks the given write response is expected and matches the given error code.
+   *
+   * @param writeResponse the write response
+   * @param codeExpected the expected error code
+   */
+  protected void checkWriteResponse(Object writeResponse, Protocol.Status.Code codeExpected) {
+    Assert.assertTrue(writeResponse instanceof RPCProtoMessage);
+
+    Object response = ((RPCProtoMessage) writeResponse).getMessage();
+    Assert.assertTrue(response instanceof Protocol.Response);
+    Assert.assertEquals(codeExpected, ((Protocol.Response) response).getStatus().getCode());
+  }
+
+  /**
+   * Checks the file content matches expectation (file length and file checksum).
+   *
+   * @param size the file size in bytes
+   * @throws IOException if it fails to check the file content
+   */
+  protected void checkFileContent(long size) throws IOException {
+    RandomAccessFile file = new RandomAccessFile(mFile, "r");
+    long checksumActual = 0;
+    long sizeActual = 0;
+
+    byte[] buffer = new byte[(int) Math.min(Constants.KB, size)];
+    int bytesRead;
+    do {
+      bytesRead = file.read(buffer);
+      for (int i = 0; i < bytesRead; i++) {
+        checksumActual += BufferUtils.byteToInt(buffer[i]);
+        sizeActual++;
+      }
+    } while (bytesRead >= 0);
+
+    Assert.assertEquals(mChecksum, checksumActual);
+    Assert.assertEquals(size, sizeActual);
+  }
+
+  /**
+   * Waits for a response.
+   *
+   * @return the response
+   */
+  protected Object waitForResponse(final EmbeddedChannel channel) {
+    return CommonUtils
+        .waitFor("Response from the channel.", new Function<Void, Object>() {
```

Comment:
NIT: I think this does not need to start with capital letter. It will follow "Timed out waiting for "